src/vs/workbench/browser/parts/editor/editorPart.ts

```diff
@@ -147,11 +151,20 @@ export class EditorPart extends Part implements IEditorPart {
 		@IThemeService themeService: IThemeService,
 		@IConfigurationService private readonly configurationService: IConfigurationService,
 		@IStorageService storageService: IStorageService,
-		@IWorkbenchLayoutService layoutService: IWorkbenchLayoutService
+		@IWorkbenchLayoutService layoutService: IWorkbenchLayoutService,
+		@IContextKeyService private contextKeyService: IContextKeyService
 	) {
 		super(id, { hasTitle: false }, themeService, storageService, layoutService);
 
 		this.registerListeners();
+
+		this.maximizedEditorGroupContext = MaximizedEditorGroupContext.bindTo(this.contextKeyService);
+
+		this.whenRestored.then(() => {
+			if (this.isGroupMaximized(this.activeGroup)) {
+				this.maximizedEditorGroupContext.set(true);
```

Comment:
I would have thought we have context key handling in the location where we also observe locked groups:

https://github.com/microsoft/vscode/blob/d58c15df07c2eb07963e456d780ad6a1a1d3173b/src/vs/workbench/browser/contextkeys.ts#L45